File: backend/Models/CustomerPhone.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace backend.Models;

public class CustomerPhone
{
    [Key]
    public Guid Id { get; set; }
    public PhoneType PhoneType { get; set; } = PhoneType.Work;
    public string PhoneNumber { get; set; } = string.Empty;
    public bool IsReceiveMessage { get; set; }
    public Guid CustomerId { get; set; }
    [JsonIgnore]
    public Customer? Customer { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

public enum PhoneType
{
    Work,
    Mobile,
    Home,
    Other
}

File: frontend/src/types/CustomerPhone.ts
import { PhoneType } from '../constants/Enumeration/CustomerEnum/CustomerPhone';

export type TCustomerPhone = {
  id: string;
  phoneType: PhoneType;
  phoneNumber: string;
  isReceiveMessage: boolean;
  customerId: string;
  createdAt: string;
  updatedAt: string;
};

export type TAddCustomerPhone = {
  phoneType: PhoneType;
  phoneNumber: string;
  isReceiveMessage: boolean;
  customerId?: string;
};

export type TUpdateCustomerPhone = {
  id: string;
  phoneType?: PhoneType;
  phoneNumber?: string;
  isReceiveMessage?: boolean;
};
